runtime/compiler/optimizer/VectorAPIExpansion.cpp

```diff
@@ -1364,29 +1510,67 @@ TR::Node *TR_VectorAPIExpansion::transformStoreToArray(TR_VectorAPIExpansion *op
       }
    else if (mode == doVectorization)
       {
-      TR::DataType vectorType = TR::DataType::createVectorType(elementType, vectorLength);
+      TR::DataType opCodeType = TR::DataType::createVectorType(elementType, vectorLength);
+      if (objType == Mask)
+          opCodeType = TR::DataType::createMaskType(elementType, vectorLength);
 
-      if (valueToWrite->getOpCodeValue() == TR::aload)
-         {
-         TR::DataType symRefType = vectorType;
+      if (valueToWrite->getOpCodeValue() == TR::aload) vectorizeLoadOrStore(opt, valueToWrite, opCodeType);
 
-         if (opt->_aliasTable[valueToWrite->getSymbolReference()->getReferenceNumber()]._objectType == Mask)
-            symRefType = TR::DataType::createMaskType(elementType, vectorLength);
+      TR::ILOpCodes op;
 
-         vectorizeLoadOrStore(opt, valueToWrite, vectorType, symRefType);
+      if (objType == Vector)
+         {
+         TR::SymbolReference *symRef = comp->getSymRefTab()->findOrCreateArrayShadowSymbolRef(opCodeType, NULL);
+         op = TR::ILOpCode::createVectorOpCode(TR::vstorei, opCodeType);
+         TR::Node::recreate(node, op);
+         node->setSymbolReference(symRef);
          }
+      if (objType == Mask)
```

Comment:
The same comment as above.
I would suggest changing this line to `else if (objType == Mask)`.